### DISCUSSION TITLE
Member is unknown when passing several items to the enum constructor

### DISCUSSION THREAD

**Discussion by davuses** [microsoft/pyright#4909] [Q&A] [answered]
While I was browsing through the documentation of the Python `enum` module, I found an [example](https://docs.python.org/3/howto/enum.html#when-to-use-new-vs-init) that is about passing several items to the constructor:
```py
from enum import Enum

class Coordinate(bytes, Enum):
    """
    Coordinate with binary codes that can be indexed by the int code.
    """
    def __new__(cls, value, label, unit):
        obj = bytes.__new__(cls, [value])
        obj._value_ = value
        obj.label = label
        obj.unit = unit
        return obj
    PX = (0, 'P.X', 'km')
    PY = (1, 'P.Y', 'km')
    VX = (2, 'V.X', 'km/s')
    VY = (3, 'V.Y', 'km/s')


print(Coordinate['PY'])
print(Coordinate.PX.label)
print(Coordinate(3))
```
When I ran the example code, there's no exception thrown, but pyright gave me this error:
```
Cannot assign member "label" for type "Coordinate"
  Member "label" is unknown

Cannot access member "label" for type "Literal[Coordinate.PX]"
  Member "label" is unknown
```
I'm confused and don't understand why the errors occur, Is there a way to make this right? thanks in advance

**Comment by erictraut** [comment, marked as answer]
The `Enum` class in Python unfortunately incorporates a bunch of very strange behaviors and capabilities that are different from all other classes in the Python stdlib. Every one of these behaviors therefore requires special casing inside of a static type checker. Not all of Enum's behaviors are supported by static type checkers. 

If you want to use static type checking, I recommend not using these exotic behaviors in the `Enum` class. Just stick to its basic functionality and you'll be fine.

**Reply by erictraut** [reply]
In this particular case, there is a workaround. You can explicitly define the `label` and `unit` variables in the class as follows:

```python
class Coordinate(bytes, Enum):
    label: str
    unit: str
    ...
```

That said, I stand by my previous recommendation. It may be fun to explore the more esoteric behaviors of `Enum`, but there are almost always better (simpler and easier-to-understand) ways to accomplish the same thing in your code.

**Reply by davuses** [reply]
Thank you for your help,  the workaround you provide does get rid of the errors. My use case is that I want to add additional attribute to enum element, which can be accessed like `Coordinate.PX.label`, but I've struggled to find a way to make it work with type checker.
